tests/test_regrid.py

```diff
@@ -123,3 +123,30 @@ def test_conservative_nans(conservative_input_data, conservative_sample_grid):
         rtol=0.002,
         atol=2e-6,
     )
+
+
+@pytest.mark.parametrize("method", ["linear", "nearest", "cubic"])
+def test_attrs_dataarray(sample_input_data, sample_grid_ds, method):
+    regridder = getattr(sample_input_data["d2m"].regrid, method)
+    assert regridder(sample_grid_ds).attrs == sample_input_data["d2m"].attrs
+
+
+def test_attrs_dataarray_conservative(sample_input_data, sample_grid_ds):
+    da_regrid = sample_input_data["d2m"].regrid.conservative(
+        sample_grid_ds, latitude_coord="latitude"
+    )
+    assert da_regrid.attrs == sample_input_data["d2m"].attrs
+
+
+@pytest.mark.parametrize("method", ["linear", "nearest", "cubic"])
+def test_attrs_dataset(sample_input_data, sample_grid_ds, method):
+    regridder = getattr(sample_input_data.regrid, method)
+    assert regridder(sample_grid_ds).attrs == sample_input_data.attrs
+
+
+def test_attrs_dataset_conservative(sample_input_data, sample_grid_ds):
+    ds_regrid = sample_input_data.regrid.conservative(
+        sample_grid_ds, latitude_coord="latitude"
+    )
+    assert ds_regrid.attrs == sample_input_data.attrs
+    assert ds_regrid["d2m"].attrs == sample_input_data["d2m"].attrs
```

Comment:
Could add an additional test for coordinate attributes, since `sample_input_data` has some for all three coordinates. However, I don't think it is strictly necessary.

Reply:
I have added tests for the coordinate attributes. It was useful, as it turns out that the `xr.interp` based methods also lost some of the attributes.